src/main/java/com/alvarium/annotators/sca/OsvBatchResponse.java

```diff
@@ -0,0 +1,87 @@
+package com.alvarium.annotators.sca;

```

Comment:
I'm sorry I must have poorly explained the typed request/response objects. 

I meant to add the 2 new types, but also keep the `OsvApiHandler` class. The response class here would only contain the serialization and de-serialization logic (if any) and the response fields. 

Here is what I imagined the `OsvApiHandler` class would look like, which may help explain my point a bit further
```java
public class OsvApiHandler implements VulnerabilityApiHandler{
    final private Gson gson = new Gson();
    final private String endpoint;

    public OsvBatchResponse() {
        this.endpoint = "https://api.osv.dev/v1/querybatch";
    }

    public OsvBatchResponse makeApiCall(Map<String,String> packages) throws ScaException {
        try {
            // Create the HTTP client and POST request
            CloseableHttpClient httpClient = HttpClients.createDefault();
            HttpPost httpPost = new HttpPost(this.endpoint);

            // Set the request body
            OsvBatchRequest req = new OsvBatchRequest(packages);
            StringEntity entity = new StringEntity(gson.toJson(req), StandardCharsets.UTF_8);
            httpPost.setEntity(entity);

            // Set the content type header
            httpPost.setHeader("Content-Type", "application/json");

            // Execute the request
            CloseableHttpResponse response = httpClient.execute(httpPost);

            // Get the response body
            HttpEntity responseEntity = response.getEntity();
            String responseBody = responseEntity != null ? EntityUtils.toString(responseEntity) : null;
            OsvBatchResponse response = new OsvBatchResponse(responseBody);

            // Close the response and the client
            response.close();
            httpClient.close();

            return response;
        } catch (IOException e) {
            throw new ScaException("Error making API call: " + e.getMessage(), e);
        }
    }


    public List<String> getVulnerabilityIds(Map<String,String> packages) throws ScaException {

        List<String> vulnerabilities = new ArrayList<>();

        OsvBatchResponse response = makeApiCall(packages);
        final List<string> vulnerabilitiyIds = response.getVulnerabilityIds();

        return vulnerabilitiesIds;
    }


}
```

I am open to discussion on this topic, but I think this gives us more separation of concerns, more re-usability and easier unit testing. what do you think ?

Reply:
> ```java
>  OsvBatchResponse response = makeApiCall(packages);
> ```

Should the deserialization be as follows:
 ```java
String response = makeApiCall(packages);
OsvBatchResponse responseBatch = gson.fromJson(response, OsvBatchResponse.class);
final List<string> vulnerabilitiyIds = responseBatch.getVulnerabilityIds();
```

Reply:
I think it's better if the de-serialization is in the constructor of the response type, that makes it so that the api handler is not concerned with serialization or deserialization. Do you have other opinions on this ?

Reply:
The only way I found/had in mind to de-serialize in the class constructor itself was something like:
```java
public OsvBatchResponse (String response) {
        OsvBatchResponse temp= gson.fromJson(response, OsvBatchResponse.class);
        this.results = temp.results;
    }

```
Which I don't think is efficient. Would be there a better way to deserialize from within the constructor on the OsvBatchResponse class instance itself?

Reply:
Oh I see... yeah I think you're right on this one, we can keep it in `makeApiCall()` then.